homeassistant/components/local_calendar/calendar.py

```diff
@@ -142,6 +125,38 @@ async def async_delete_event(
         await self._async_store()
         await self.async_update_ha_state(force_refresh=True)
 
+    async def async_update_event(
+        self,
+        uid: str,
+        event: dict[str, Any],
+        recurrence_id: str | None = None,
+        recurrence_range: str | None = None,
+    ) -> None:
+        """Update an existing event on the calendar."""
+        new_event = _parse_event(event)
+        range_value: Range = Range.NONE
+        if recurrence_range == Range.THIS_AND_FUTURE:
+            range_value = Range.THIS_AND_FUTURE
+        EventStore(self._calendar).edit(
+            uid,
+            new_event,
+            recurrence_id=recurrence_id,
+            recurrence_range=range_value,
+        )
+        await self._async_store()
+        await self.async_update_ha_state(force_refresh=True)
+
+
+def _parse_event(event: dict[str, Any]) -> Event:
+    """Parse an ical event from a home assistant event dictionary."""
+    if rrule := event.get(EVENT_RRULE):
+        event[EVENT_RRULE] = Recur.from_rrule(rrule)
+    try:
+        return Event.parse_obj(event)
+    except ValidationError as err:
+        _LOGGER.debug("Error parsing event input fields: %s (%s)", event, str(err))
+        raise vol.Invalid("Error parsing event input fields") from err
```

Comment:
What catches `vol.Invalid`?

Reply:
My impression is the http view does this:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/http/view.py#L137

This goes back to the debate we were having about `ValueError` vs `vol.Invalid`.  You last suggested we take it to an architecture issue and i haven't followed up on this.

Reply:
Ok. It's a bit weird to raise `vol.Invalid` as we're not inside a voluptuous schema. But I'll guess we'll have to keep it like this until that's generally cleaned up from the views and an alternative added.